common/s3util.cpp

```diff
@@ -124,6 +126,21 @@ GetObjectsResponse S3Util::getObjects(
   }
 }
 
+GetObjectMetadataResponse S3Util::getObjectMetadata(const string &key) {
+  HeadObjectRequest headObjectRequest;
+  headObjectRequest.SetBucket(bucket_);
+  headObjectRequest.SetKey(key);
+  auto headObjectResult = s3Client.HeadObject(headObjectRequest);
+  map<string, string> metadata;
+  if (headObjectResult.IsSuccess()) {
+    metadata = headObjectResult.GetResult().GetMetadata();
+    return GetObjectMetadataResponse(metadata, "");
```

Comment:
NIT: should we return something indicating the request is successful instead of an empty string?